File: Arrays/sorting/SelectionSort.cpp
#include<iostream>
#include<vector>
using namespace std;
int main(){

vector<int>arr;
int n;
cin>>n;
//taking input for inserting values in array
for(int i=0;i<n;i++){
    int c;

    cin>>c;
    arr.push_back(c);
}


//Selection Sort 

for(int i=0;i<n-1;++i){
    int min_idx=i;

    for(int j=i+1;j<n;++j){
        if(arr[min_idx]>arr[j]){
               min_idx=j;
        }
    }
  if(min_idx!=i){
    swap(arr[i],arr[min_idx]);
  }
}

}

File: Arrays/sorting/SelectionSort.py
# Python implementation of Selection Sort

# Take input for the number of elements
n = int(input())

# Create an empty list to store the elements
arr = []

# Take input for each element and append to the list
for _ in range(n):
    c = int(input())
    arr.append(c)

# Selection Sort
for i in range(n-1):
    min_idx = i
    for j in range(i+1, n):
        if arr[min_idx] > arr[j]:
            min_idx = j
    if min_idx != i:
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

# Print the sorted list
print(arr)